R/createFigure3.R

```diff
@@ -0,0 +1,221 @@
+                           value=numeric(),
+                           units = character(),
+                           wqx.id = character(),
+                           dec_lat_va = numeric(),
+                           dec_lon_va = numeric(),
+                           StateCode = character(),
+                           ActivityTypeCode = character(),
+                           ActivityMediaName = character(),
+                           ResultStatusIdentifier = character())
+  }
+  
+  return(out.data)
+
+}
+
+
+
+library(dplyr)
+library(dataRetrieval)
+library(lubridate)
+library(ggplot2)
+
+characteristicNames = c("Depth, Secchi disk depth", 
+                        "Depth, Secchi disk depth (choice list)", 
+                        "Secchi Reading Condition (choice list)", 
+                        "Secchi depth", 
+                        "Water transparency, Secchi disc")
+siteTypes = "Lake, Reservoir, Impoundment"
+
+# To get the full data set used to produce the figure in the text:
+# This takes roughly 15 minutes
+# get_us_secchi(outfile="all_secchi_usa_long.rds",
+#               characteristicNames,
+#               siteTypes,
+#               stateCode = "All",
+#               startDateLo = '1900-01-01',
+#               startDateHi = '2016-01-01',
+#               stride = "20 years")
+
+# To get the small subset to test the workflow:
+# This takes roughly 3 minutes
+get_us_secchi(outfile="sub_secchi_AL_MN.rds",
+              characteristicNames,
+              siteTypes,
+              stateCode = c("01","27"), #AL and MN
+              startDateLo = '2000-01-01', 
+              startDateHi = '2016-01-01',
+              stride = "20 years")
+
+infile <- "all_secchi_usa_long.rds"
+secchi.data <- readRDS(infile)
+
+regions <- data.frame(STATE_NAME = c('Montana', 'Wyoming', 'Idaho', 'Washington', 'Oregon', 'California', 'Nevada',
+                                     'Arizona', 'New Mexico', 'Colorado', 'Utah'), group='West', stringsAsFactors = FALSE) %>% 
+  rbind(data.frame(STATE_NAME=c('Ohio', 'Indiana', 'Illinois', 'Wisconsin', 'Missouri', 'Iowa', 'Minnesota', 'Kansas',
+                                'Nebraska', 'South Dakota', 'North Dakota', 'Michigan'), group='Midwest', stringsAsFactors = FALSE)) %>% 
+  rbind(data.frame(STATE_NAME=c('Maine', 'New Hampshire', 'Vermont', 'Massachusetts', 'Rhode Island', 'Connecticut', 
+                                'New Jersey', 'Pennsylvania', 'New York'), group='Northeast', stringsAsFactors = FALSE)) %>% 
+  rbind(data.frame(STATE_NAME=c('Florida', 'Georgia', 'Louisiana', 'Arkansas', 'Oklahoma', 'Texas', 'South Carolina', 
+                                'North Carolina', 'Virginia', 'Kentucky', 'Tennessee', 'West Virginia', 'Maryland', 
+                                'Delaware', 'Alabama', 'Mississippi'), group='South', stringsAsFactors = FALSE))
+
+secchi.filtered <- secchi.data %>%
+  filter(ActivityTypeCode %in% c("Field Msr/Obs","Sample-Routine",
```

Comment:
this failed because 'ActivityTypeCode' is not a column name in secchi.data 
```r
names(secchi.data)
[1] "Date"       "wqx.id"     "secchi"     "dec_lat_va" "dec_lon_va" "StateCode" 
```